subsys/bluetooth/mesh/scheduler_srv.c

```diff
@@ -47,6 +48,23 @@ static bool set_second(struct tm *sched_time,
 		       struct bt_mesh_schedule_entry *entry,
 		       struct bt_mesh_time_srv *srv);
 
+static int store(struct bt_mesh_scheduler_srv *srv, uint8_t idx)
+{
+	char name[2] = {0};
+
+	if (!IS_ENABLED(CONFIG_BT_SETTINGS)) {
+		return 0;
+	}
+
+	name[0] = ENTRY_BASE_KEY + idx;
+	BT_DBG("name %s, idx %d", name, idx);
+
+	struct bt_mesh_schedule_entry data = srv->sch_reg[idx];
```

Comment:
No need for this stack variable, passing a pointer directly into the sch_reg to data_store is perfectly fine, and saves some stack space.